### PR TITLE
if let instead of match, rm ref

### PR DESCRIPTION
**Pull request by burrbull**

### CODE REVIEW

**Comment by burrbull** [comment]
bors try

File: src/svd/enumeratedvalue.rs [hunk 1]
```diff
@@ -75,31 +75,21 @@ impl Encode for EnumeratedValue {
             text: None,
         };
         // FIXME: Use if let some pattern here
```
Comment (Disasm) [inline src/svd/enumeratedvalue.rs]: Remove the comment?

File: src/svd/register.rs [hunk 2]
```diff
@@ -25,9 +25,9 @@ impl Deref for Register {
     type Target = RegisterInfo;
 
     fn deref(&self) -> &RegisterInfo {
-        match *self {
-            Register::Single(ref info) => info,
-            Register::Array(ref info, _) => info,
+        match &*self {
```
Comment (Disasm) [inline src/svd/register.rs]: Is `&*` required here?

File: src/svd/enumeratedvalue.rs (hunk 1 above)
Comment (burrbull) [reply src/svd/enumeratedvalue.rs]: Done.

File: src/svd/register.rs (hunk 2 above)
Comment (burrbull) [reply src/svd/register.rs]: Removed.

**Comment by Disasm** [comment]
Please, do not force-push changes after review, it will make it easier to review incrementally (instead of starting from scratch).

**Pull request merged**